### PR TITLE
chore(storybook): switch to static Clever App

### LINKED ISSUE #1280: Storybook: static server crashes every now and then
## Context

For our prod storybook instance (https://www.clever-cloud.com/doc/clever-components/?path=/docs/readme--docs), we rely on [`node-static`](https://github.com/cloudhead/node-static) as a simple static Node.js server.

It seems that every now and then, the instance crashes and reboots with the following error: 

```sh
2024-12-11T13:57:18.285Z node:_http_server:345
2024-12-11T13:57:18.285Z     throw new ERR_HTTP_HEADERS_SENT('write');
2024-12-11T13:57:18.285Z     ^
2024-12-11T13:57:18.285Z Error [ERR_HTTP_HEADERS_SENT]: Cannot write headers after they are sent to the client
2024-12-11T13:57:18.285Z     at ServerResponse.writeHead (node:_http_server:345:11)
2024-12-11T13:57:18.285Z     at Server.finish (/home/bas/app_91d78df5-c3dd-4c7f-b09a-8628e7e54860/node_modules/node-static/lib/node-static.js:125:17)
2024-12-11T13:57:18.285Z     at finish (/home/bas/app_91d78df5-c3dd-4c7f-b09a-8628e7e54860/node_modules/node-static/lib/node-static.js:170:14)
2024-12-11T13:57:18.285Z     at /home/bas/app_91d78df5-c3dd-4c7f-b09a-8628e7e54860/node_modules/node-static/lib/node-static.js:337:13
2024-12-11T13:57:18.285Z     at streamFile (/home/bas/app_91d78df5-c3dd-4c7f-b09a-8628e7e54860/node_modules/node-static/lib/node-static.js:382:13)
2024-12-11T13:57:18.285Z     at ReadStream.<anonymous> (/home/bas/app_91d78df5-c3dd-4c7f-b09a-8628e7e54860/node_modules/node-static/lib/node-static.js:375:17)
2024-12-11T13:57:18.285Z     at ReadStream.emit (node:events:519:28)
2024-12-11T13:57:18.285Z     at emitCloseNT (node:internal/streams/destroy:147:10)
2024-12-11T13:57:18.285Z     at process.processTicksAndRejections (node:internal/process/task_queues:81:21) {
2024-12-11T13:57:18.285Z   code: 'ERR_HTTP_HEADERS_SENT'
2024-12-11T13:57:18.285Z }
```

TLDR; the server tries to send a response but it has already sent one.

There is an open issue about this error on the `node-static` repo: https://github.com/cloudhead/node-static/issues/241

I couldn't determine what types of requests triggered the error so I still need to dig a little bit but we may need to change the static server we use. 
`node-static` hasn't been updated in 6 years, which is not necessarily an issue but even if it doesn't pull many dependencies, the ones it pulls are fairly outdated.

### PR DESCRIPTION
**Pull request by florian-sanders-cc**
Fixes #1280

## What does this PR do?

- Removes the `node-static` dependency,
- Removes the `start` npm script,
- Adds `.htaccess` to be used by the static Clever App (Apache server).

## How to review?

- Check the code,
- The preview shouldn't be impacted,
- Run the `npm run storybook:build` locally and check that the `storybook-static` directory contains a `.htaccess`.

## Things to discuss

To make Apache handle the SPA correctly & specify cache rules, we need a `.htaccess` file.
This `.htaccess` file needs to be placed at the root of the directory served by the server (`CC_WEBROOT`) which is `storybook-static`.

We cannot simply put the `.htaccess` in `storybook-static` ourselves because this directory is removed & created by the storybook build command (+ it's git ignored anyway).
We have two options:
- make the build command copy the `.htaccess` file in `storybook-static` (the option I chose in this PR),
- or add a `CC_POST_BUILD_HOOK` to move the `.htaccess` file in the `storybook-static` directory only when building on Clever Cloud.

The option I've chosen has 1 advantage:  you cannot forget to add this step, it's part of the build process.
It also has 1 disadvantage: we send `.htaccess` files even for previews which is useless because they are served by Cellar.

### CODE REVIEW

**Review by pdesoyres-cc — APPROVED** [review]
LGTM. Well done.

The option you chose for the `.htaccess` file handling is ok for me.

**Review by roberttran-cc — APPROVED** [review]
Thanks for handling this! And I agree with the choice you made for the `.htaccess`.
